chain/mempool.go

```diff
@@ -347,12 +376,21 @@ func (m *mempool) UpdateMempoolTxes(txids []*chainhash.Hash) []*wire.MsgTx {
 	// Set all mempool txs to false.
 	m.UnmarkAll()
 
+	// TODO(yy): let the OS manage the number of goroutines?
```

Comment:
The RPC requests are already processed sequentially in `btcd` tho, so it won't have any effect here on that front. Unless we are concerning about the overhead of creating and destroying goroutines, I think it's better to let the runtime scheduler to handle them. Also `GOMAXPROCS` is already default to `runtime.NumCPU` 

> it probably still makes sense to limit the number of goroutines to prevent clogging on slower systems.

Have we experienced this before?

Reply:
> Have we experienced this before?

Yes, at least I think that in certain situations (e.g. during itests) when we ran too many things in parallel, some goroutines would just be starved (didn't get any CPU time). Not sure if that gets worse with the number of goroutines or just with the number of processes running on a system.

Reply:
Yes. So when the CPU is already struggling, goroutines can get starved (at least it looked like it). But I'm not sure if that gets worse with the number of goroutines there are in total or whether that doesn't have any direct impact. Just some thoughts/observations to possibly help the decision here.